src/app/config.rs

```diff
@@ -145,6 +145,9 @@ pub struct Config {
     pub dns_canonicalize_timeout: Duration,
 
     pub h2_settings: H2Settings,
+
+    /// When set, tap is not served
+    pub tap_disabled: bool,
```

Comment:
What if, instead of this flag, we make the `control_listener` hold an `Option` so that it only exists when tap is enabled?